### DISCUSSION TITLE
How to save generated wavesurfer as wav and mp3 file?

### DISCUSSION THREAD

**Discussion by UchihaYuki** [katspaugh/wavesurfer.js#3521] [Q&A] [answered]
I generate the 2nd wavesurfer from the selected region of the first wavesurfer.

The screenshot:
![image](https://github.com/katspaugh/wavesurfer.js/assets/38900426/b7f269be-1466-4dfa-983f-2e6b8246098e)

Is there a way to export the 2nd wavesurfer as both wav and mp3 files on the user's local disk?

**Comment by katspaugh** [comment]
[`getDecodedData`](https://wavesurfer.xyz/docs/classes/wavesurfer.default#getDecodedData)

**Reply by UchihaYuki** [reply]
I try to reference [this web page](https://russellgood.com/how-to-convert-audiobuffer-to-audio-file/#render-as-wav) to turn the generated wavesurfer to a wav file.

But I found the `this.generatedWavesurfer.getDecodedData()` returns null.

I guess that's why [it refuses to play](https://github.com/katspaugh/wavesurfer.js/discussions/3520).

**Reply by katspaugh** [reply]
You should call it on ready.

PS Please try not to create several threads

**Reply by UchihaYuki** [reply, marked as answer]
I guess I found the reason. The region selected is too short. And browsers seem to have a problem decoding those short wav files.
https://stackoverflow.com/a/73127796

I tried a long enough region and it works now.

> You should call it on ready.

I'm pretty sure it's ready.

> Please try not to create several threads

Do you mean I should use only one wavesurfer instance? What if I need several waveforms?

**Reply by katspaugh** [reply]
I’m talking about discussion threads. You opened two on the same topic.

Anyway, glad you figured it out. Good luck!

**Reply by UchihaYuki** [reply]
I thought they focused on two different things. It might help when people search.
